Mammoth/TSE/CShockwaveHitTest.cpp

```diff
@@ -189,13 +189,16 @@ void CShockwaveHitTest::Update (SEffectUpdateCtx &Ctx, const CVector &vPos, Metr
 	TArray<SHitData> SegHit;
 	SegHit.InsertEmpty(m_Segments.GetCount());
 
-	//	Loop over all objects in the system
+	const CSpaceObjectGrid &Grid = Ctx.pSystem->GetObjectGrid();
+	SSpaceObjectGridEnumerator i;
+	Grid.EnumStart(i, vUR, vLL, 0);
 
-	for (i = 0; i < Ctx.pSystem->GetObjectCount(); i++)
+	//	Loop over all objects in and near the grid rectangle
+
+	while (Grid.EnumHasMore(i))
 		{
-		CSpaceObject *pObj = Ctx.pSystem->GetObject(i);
+		CSpaceObject *pObj = Grid.EnumGetNextFast(i);
```

Comment:
```suggestion
		CSpaceObject *pObj = Grid.EnumGetNext(i);
```
Most collision detection appears to use `EnumGetNext` rather than the "Fast" version. The code comments done explain why one should be used over the other.. However it looks like the "Fast" version drops the InBox / destroyed check, so probably best to stick with the standard version here.

Note - if going with `EnumGetNext` we should be able to remove the IsDestroyed check just below